tests/lib/prepare.sh

```diff
@@ -509,22 +512,9 @@ prepare_all_snap() {
     setup_systemd_snapd_overrides
 
     # Snapshot the fresh state (including boot/bootenv)
-    if [ ! -f "$SPREAD_PATH/snapd-state.tar.gz" ]; then
-        # we need to ensure that we also restore the boot environment
-        # fully for tests that break it
-        BOOT=""
-        if ls /boot/uboot/*; then
-            BOOT=/boot/uboot/
-        elif ls /boot/grub/*; then
-            BOOT=/boot/grub/
-        else
-            echo "Cannot determine bootdir in /boot:"
-            ls /boot
-            exit 1
-        fi
-
+    if [ ! -d $SNAPD_STATE_PATH ]; then
         systemctl stop snapd.service snapd.socket
-        tar cf "$SPREAD_PATH/snapd-state.tar.gz" /var/lib/snapd $BOOT /etc/systemd/system/snap-*core*.mount
+        save_all_snap_state
```

Comment:
With the check for non-empty "$boot_path" this will fail as expected if boot path cannot be determined.